### PR TITLE
IAC-1124: add iacm workspaces

### PR DESCRIPTION
**Pull request by dwarren-dev**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

**Comment by dwarren-dev** [comment]
trigger subcategoryfieldcheck

File: .changelog/720.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+```release-note:new-resource
+  harness_platform_workspace: Added workspace resource.
+```
```
Comment (rathodmeetsatish) [inline .changelog/720.txt:3]: changelog file name should be corresponding to the pr umber i this case it is 726

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
added comment

**Review by allofthesepeople — APPROVED** [review]

**Review by nassergonzalez — DISMISSED** [review]

File: docs/data-sources/platform_workspace.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_workspace Data Source - terraform-provider-harness"
+subcategory: ""
```
[the remark this answers is not in the document]
Comment (dwarren-dev) [reply docs/data-sources/platform_workspace.md]: @rathodmeetsatish i made these changes, can you re-review?

File: .changelog/726.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+```release-note:new-resource
+  harness_platform_workspace: Added workspace resource.
+```
+
+```release-note:new-data-source
```
Comment (rathodmeetsatish) [inline .changelog/726.txt:5]: we only have 3 options here new-resource, enhancement, fix

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

File: .changelog/726.txt (hunk 3 above)
Comment (dwarren-dev) [reply .changelog/726.txt]: @rathodmeetsatish thanks, fixed this in my latest commit

**Review by rathodmeetsatish — DISMISSED** [review]

**Review by nassergonzalez — APPROVED** [review]

**Pull request merged**